### DISCUSSION TITLE
How to disable initial one-way setting sync?

### DISCUSSION THREAD

**Discussion by liuzhenqi77** [community/community#41889] [Codespaces] [answered]
### Select Topic Area

Question

### Body

According to [this thread](https://github.com/community/community/discussions/27269)

> When you create and connect to a new codespace in the web browser, a one-way sync is performed to pull all your settings into your new codespace once. If all you want is to ensure that your codespaces contain your settings, you don't need to enable settings sync at all.

I have some plugins synced across my local machines (so I prefer not to use `settingsSync.ignoredExtensions`), but they are slow to load and not relevant for the codespaces.
I was wondering how to disable this one-way sync, and only use settings and plugins from `devcontainer.json`?

**Comment by smr001** [comment]
Can some help me out on how to set a payment API? CONTACT ME AT +1 972 994 6843
meghanweidnerrichard@gmail.com

**Comment by erikaellr** [comment]
Hi @liuzhenqi77, we're thrilled you're using Codespaces! Unfortunately, what you'd like to do is not currently supported. There currently isn't a way to disable that initial sync to a codespace, and while there is a user setting `"github.codespaces.defaultExtensions": []` that will allow you to specify a list of extensions that should be installed in codespaces even if they are not in your locally installed and synced extensions, there is no setting that will allow you to specify a list of extensions that are excluded from that initial sync to a codespace.

We are aware that this initial sync to a codespace, whether Settings Sync is enabled or not, has been a point of confusion or frustration for many user. We can't guarantee that every bit of feedback sees action, but we are paying attention and looking for solutions. Thanks for adding your experience on this feature.

**Reply by JustinGrote** [reply]
What do you recommend if I want to test a codespace to make sure it operates correctly for a new user who wants to work on my project? This is a very common ask.

**Comment by MSWS** [comment]
Bit bizarre that this isn't a feature. I hope it gets added soon. Thanks for the indepth and quick response nonetheless.

**Comment by spaltrowitz** [comment, marked as answer]
Hi @liuzhenqi77 @JustinGrote @MSWS thanks for sharing this question. A related feature has now been shipped! 🚢 https://github.blog/changelog/2023-05-04-codespaces-settings-sync-updates/

**Reply by Arlen22** [reply]
> Hi @liuzhenqi77 @JustinGrote @MSWS thanks for sharing this question. A related feature has now been shipped! 🚢 https://github.blog/changelog/2023-05-04-codespaces-settings-sync-updates/

I have no idea what this fixes. I have no repos set to sync and yet it still syncs, entirely messing up my codespace.

**Comment by Arlen22** [comment]
Easiest way seems to be to make a new profile. https://code.visualstudio.com/docs/editor/profiles
